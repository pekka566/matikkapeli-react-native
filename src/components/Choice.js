import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { View, Text, StyleSheet } from 'react-native';
import Button from 'apsl-react-native-button';

const Choice = props => (
  <View style={styles.container}>
    <Button
      style={styles.buttonStyle}
      onPress={() => {
        props.handlePress(props.value);
      }}
      isDisabled={props.buttonDisabled}
    >
      <View style={styles.buttonView}>
        <Text style={styles.buttonText}>{props.value}</Text>
      </View>
    </Button>
  </View>
);

Choice.defaultProps = {
  buttonDisabled: false
};

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    marginLeft: '2%',
    marginRight: '2%'
  },
  buttonStyle: {
    backgroundColor: '#9b59b6',
    borderColor: '#333',
    borderWidth: 3,
    borderRadius: 10
  },
  buttonView: {
    width: '100%',
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'center'
  },
  buttonText: {
    textAlign: 'center', // <-- the magic
    fontWeight: 'bold',
    fontSize: 36,
    color: '#fff',
    flex: 1,
    lineHeight: 100,
    textShadowOffset: { width: 5, height: 5 },
    textShadowRadius: 6
  }
});

export default Choice;
