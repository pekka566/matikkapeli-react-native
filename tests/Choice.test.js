import React from 'react';
import { shallow } from 'enzyme';
import Choice from '../src/components/Choice';

it('renders without crashing', () => {
  const handlePressMock = jest.fn();
  const wrapper = shallow(
    <Choice
      buttonDisabled={false}
      value={'Press mee!'}
      handlePress={handlePressMock}
    />
  );
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('Button').length).toBe(1);
  expect(wrapper.find('Text').length).toBe(1);
  expect(wrapper.find('View').length).toBe(2);
  expect(
    wrapper
      .find('Button')
      .first()
      .props().isDisabled
  ).toBe(false);

  wrapper
    .find('Button')
    .first()
    .props()
    .onPress();
  expect(handlePressMock.mock.calls.length).toBe(1);

  wrapper.setProps({
    buttonDisabled: true
  });
  expect(
    wrapper
      .find('Button')
      .first()
      .props().isDisabled
  ).toBe(true);

  wrapper
    .find('Button')
    .first()
    .props()
    .onPress();
  expect(handlePressMock.mock.calls.length).toBe(2); // it is still possible to press in test even the button is disabled!
});
