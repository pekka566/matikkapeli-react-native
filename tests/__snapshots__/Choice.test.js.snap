// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginLeft": "2%",
      "marginRight": "2%",
    }
  }
>
  <Button
    isDisabled={false}
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "#9b59b6",
        "borderColor": "#333",
        "borderRadius": 10,
        "borderWidth": 3,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "flex": 1,
            "fontSize": 30,
            "fontWeight": "bold",
            "lineHeight": 100,
            "textAlign": "center",
            "textShadowOffset": Object {
              "height": 5,
              "width": 5,
            },
            "textShadowRadius": 6,
          }
        }
      >
        Press mee!
      </Text>
    </View>
  </Button>
</View>
`;
